### ISSUE TITLE
Problem when ending on attack period 

### ISSUE THREAD

**Issue by marota** [marota/Oracle4Grid#41]
When we end the episode on an attack period, the "end" node is not properly connected to the reward graph

For instance when running 70 iterations

• new attack nodes at t69 are connected to « init » node but not « end » node
        ◦ some are duplicates of old nodes connected to end node .. !
• old nodes at t69 are connected to « end » node but not « init » node

```
oracle4grid
-d 1 -f oracle4grid/ressources/actions/neurips_track1/ExpertActions_Track1_action_list_score4.json
-e /home/L2RPN_Codalab_Competition_Bundle/competition_codalab/L2RPN_neurips2020_track1/input_data_val
-c apr19.1
-as 858758473
-es 858758473

```

**Comment by marota** [comment]
Should review this function for that
https://github.com/marota/Oracle4Grid/blob/79ca824e97cd8f954670bfc305008421e8d905cc/oracle4grid/core/graph/graph_generator.py#L117

Also this might highlight some duplicated nodes in the period of attacks in the dataframe

**Comment by marota** [comment]
Changing 
```
edges_ex_t =[str(name) + '_t' + str(max_iter - 1) for name in ordered_names]
```

by
```
edges_ex_t = [node for node in uniques_ex if '_t' + str(max_iter - 1) in node]
```
solves it.

You can check that 

```
g_und = graph.to_undirected()
nx.number_connected_components(g_und)
```
Gives 1 fully connected component instead of two before

**Comment by marota** [comment]
Solved in 56afd21026a56e560d493275e3d7b9438d18e7f8
